packages/nimble-components/src/table-column/base/index.ts

```diff
@@ -98,4 +104,14 @@ export abstract class TableColumn<
             this.columnInternals.currentSortIndex = this.sortIndex;
         }
     }
+
+    protected getTypedValidator<T extends ColumnValidator<[]>>(
```

Comment:
As an approach to giving feedback sooner (will try and discuss next meeting), I'll give the high level feedback that I'd like to try and find an alternative to this pattern. It is described in the PR description why the direction was chosen but still feels weird. But review is stalled because haven't had the chance to play with alternatives yet / have ideas to propose yet.

Reply:
Since it's not a high priority tech debt issue and haven't had a chance to look closer, assigning to myself and moving to draft to take another look when there is more bandwidth